pkg/wallet/wallet_test.go

```diff
@@ -1,8 +1,103 @@
 package wallet
 
 import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	walletPath = "testWallet"
 )
 
 func TestNewWallet(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+	require.NotNil(t, wallet)
+}
+
+func TestNewWalletFromFile_Negative_EmptyFile(t *testing.T) {
+	_ = checkWalletConstructor(t)
+	walletFromFile, err2 := NewWalletFromFile(walletPath)
+	require.Errorf(t, err2, "EOF")
+	require.Nil(t, walletFromFile)
+}
+
+func TestNewWalletFromFile_Negative_NoFile(t *testing.T) {
+	_, err := NewWalletFromFile(walletPath)
+	require.Errorf(t, err, "open testWallet: no such file or directory")
+}
+
+func TestCreateAccount(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+
+	errAcc := wallet.CreateAccount("testName", "testPass")
+	assert.NoError(t, errAcc)
+	accounts := wallet.Accounts
+	assert.Len(t, accounts, 1)
+}
+
+func TestAddAccount(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+
+	wallet.AddAccount(&Account{
+		privateKey:   nil,
+		publicKey:    nil,
+		wif:          "",
+		Address:      "",
+		EncryptedWIF: "",
+		Label:        "",
+		Contract:     nil,
+		Locked:       false,
+		Default:      false,
+	})
+	accounts := wallet.Accounts
+	assert.Len(t, accounts, 1)
+}
+
+func TestPath(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+
+	path := wallet.Path()
+	assert.NotEmpty(t, path)
+}
+
+func TestSave(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+
+	err := wallet.Save()
+	assert.NoError(t, err)
+}
+
+func TestJSONMarshallUnmarshal(t *testing.T) {
+	wallet := checkWalletConstructor(t)
+
+	bytes, err := wallet.JSON()
+	assert.NoError(t, err)
+	assert.NotNil(t, bytes)
+
+	unmarshalledWallet := &Wallet{}
+	errUnmarshal := json.Unmarshal(bytes, unmarshalledWallet)
+
+	assert.NoError(t, errUnmarshal)
+	assert.Equal(t, wallet.Version, unmarshalledWallet.Version)
+	assert.Equal(t, wallet.Accounts, unmarshalledWallet.Accounts)
+	assert.Equal(t, wallet.Scrypt, unmarshalledWallet.Scrypt)
+}
+
+func checkWalletConstructor(t *testing.T) *Wallet {
+	file, err := ioutil.TempFile("", walletPath)
+	assert.NoError(t, err)
+	wallet, err := NewWallet(file.Name())
+	defer removeWallet(t, file.Name())
+	require.NoError(t, err)
+	return wallet
+}
+
+func removeWallet(t *testing.T, walletPath string) {
+	err := os.RemoveAll(os.TempDir() + "/" + walletPath)
```

Comment:
Why do you add a `TempDir()` here? `file.Name()` already returns an absolute path to your file, here you're deleting something like `/tmp/tmp/testWalletXXXXX`.